python_training/mimic.py

```diff
@@ -0,0 +1,186 @@
+    Each member in the list would be a ListMember object.
+    The MyList object holds a direct pointer to the first and last member in the list, and
+    each ListMember holds a direct pointer to the proceeding ListMember object.
+    """
+    def __init__(self):
+        """
+        Initiates a MyList object.
+        """
+        self.first_member = None
+        self.last_member = None
+        self.list_length = 0
+
+    def append(self, value):
+        """
+        Creates a new ListMember and adds it to the list.
+
+        :param value: The value that would be held in the ListMember.
+        """
+        if self.first_member is None:
+            self.first_member = ListMember(value)
+            self.last_member = self.first_member
+        else:
+            self.last_member.next_member = ListMember(value)
+            self.last_member = self.last_member.next_member
+        self.list_length += 1
+
+    def __len__(self)-> int:
+        """
+        Returns the length of the list.
+
+        :return: The length of the list.
+        """
+        return self.list_length
+
+    def __iter__(self):
+        """
+        Initiates and returns an Iterator object that allows iterating over the list members.
+
+        :return: A Iterator object.
+        """
+        return ListIterator(self)
+
+    def remove(self, value_to_remove):
+        """
+        Removes the first ListMember that holds the given value.
+
+        :param value_to_remove: A value that would be removed from the list.
+        """
+        if self.list_length == 0:
+            raise ValueError('MyList.remove(x): x not in list')
+        for member in self:
+            if member is self.first_member and member.value == value_to_remove:
+                if self.list_length == 1:
+                    self.first_member = None
+                    self.last_member = None
+                else:
+                    self.first_member = self.first_member.next_member
+                self.list_length -= 1
+                break
+
+            if member.next_member.value == value_to_remove:
+                if member.next_member is self.last_member:
+                    self.last_member = member
+                    member.next_member = None
+                    self.list_length -= 1
+                    break
+                else:
+                    member.next_member = member.next_member.next_member
+                    self.list_length -= 1
+                    break
+            if member == self.last_member:
+                raise ValueError('MyList.remove(x): x not in list')
+
+    def __str__(self)-> str:
+        """
+        Returns a textual representation of the list.
+
+        :return: A string representing the list.
+        """
+        base_string = '[ '
+        for member in self:
+            base_string += '{}, '.format(member.value) if member is not self.last_member else '{}'.format(member.value)
+        base_string += ' ]'
```

Comment:
This can be a part of the `for`, since you already have special treatment for the last element.